src/Lit/Lit.fs

```diff
@@ -45,6 +45,20 @@ type CSSResult =
     abstract cssText: string
     abstract styleSheet: CSSStyleSheet
 
+// https://lit.dev/docs/api/controllers/#ReactiveController
+type ReactiveController =
```

Comment:
One thing to have in mind is that all of these are [actually optional](https://github.com/lit/lit/blob/main/packages/reactive-element/src/reactive-controller.ts#L53-L84) so if someone wants to implement these in F# F# will force them to implement the 4 members do you think that can become an issue?

Reply:
Unfortunately there's no good solution for this in F#. If we mark these members as `(unit -> unit) option` you still need to implement them. And the compiler will translate them as getters which will also cause issues:

```fsharp
type ReactController =
    abstract hostConnected: (unit -> unit) option
    abstract hostDisconnected: (unit -> unit) option
    abstract hostUpdate: (unit -> unit) option
    abstract hostUpdated: (unit -> unit) option

type MyController() =
    interface ReactController with
        member _.hostConnected = Some(fun () -> printfn "Connected")
        member _.hostDisconnected = None
        member _.hostUpdate = None
        member _.hostUpdated = None
```

Becomes:

```js
export class MyController {
    constructor() {
    }
    get hostConnected() {
        return () => {
            toConsole(printf("Connected"));
        };
    }
    get hostDisconnected() {
        return void 0;
    }
    get hostUpdate() {
        return void 0;
    }
    get hostUpdated() {
        return void 0;
    }
}
```

The good news is that being methods returning unit, it's very simple to have empty implementations:

```fsharp
type ReactController =
    abstract hostConnected: unit -> unit
    abstract hostDisconnected: unit -> unit
    abstract hostUpdate: unit -> unit
    abstract hostUpdated: unit -> unit

type MyController() =
    interface ReactController with
        member _.hostConnected() = printfn "Connected"
        member _.hostDisconnected() = ()
        member _.hostUpdate() = ()
        member _.hostUpdated() = ()
```

Reply:
Yeah I agree, in the worst case scenario one can always return an anonymous object 
```fsharp
{| hostConnected = fun () -> ... |}
```
and ignore the rest